packages/checkout-storefront/src/sections/PaymentSection/AdyenDropIn.tsx

```diff
@@ -0,0 +1,193 @@
+import {
+  createDropInAdyenPayment,
+  createDropInAdyenSession,
+} from "@/checkout-storefront/fetch/requests";
+import { useCheckout, useFetch } from "@/checkout-storefront/hooks";
+import { useAppConfig } from "@/checkout-storefront/providers/AppConfigProvider";
+import AdyenCheckout from "@adyen/adyen-web";
+import { CardElementData } from "@adyen/adyen-web/dist/types/components/Card/types";
+import DropinElement from "@adyen/adyen-web/dist/types/components/Dropin";
+import { memo, useCallback, useEffect, useLayoutEffect, useRef, useState } from "react";
+import { PaymentResponse as AdyenPaymentResponse } from "@adyen/api-library/lib/src/typings/checkout/paymentResponse";
+import { AdyenDropInCreateSessionResponse } from "checkout-common";
+
+type AdyenCheckoutInstance = Awaited<ReturnType<typeof AdyenCheckout>>;
+
+const useEvent = <Args extends unknown[], R>(handler: (...args: Args) => R) => {
+  const handlerRef = useRef<null | ((...args: Args) => R)>(null);
+
+  useLayoutEffect(() => {
+    handlerRef.current = handler;
+  });
+
+  return useCallback((...args: Args) => {
+    return handlerRef.current?.(...args) as R;
+  }, []);
+};
```

Comment:
Why do we need to do this sort of shenanigans? 🤔 Could you perhaps add some comment to help me here? 🙏🏻 

Another thing - maybe this should be in a separate file in `utils` folder?

Reply:
https://github.com/reactjs/rfcs/blob/useevent/text/0000-useevent.md